internal/hardware/validator.go

```diff
@@ -576,3 +575,41 @@ func compileDiskReasonTemplate(template string, wildcards ...interface{}) *regex
 	}
 	return tmp
 }
+
+// preciseHumanizeBytes formats a byte size using SI units (KB, MB, GB, TB).
+// It displays one decimal place for non-integer values and rounds up to avoid
+// under-reporting the capacity in user-facing messages.
+func preciseHumanizeBytes(bytes uint64) string {
+	const (
+		KB = 1000
+		MB = 1000 * KB
+		GB = 1000 * MB
+		TB = 1000 * GB
+	)
+
+	switch {
+	case bytes >= TB:
+		tb := float64(bytes) / float64(TB)
+		return formatUnitRoundedUpOneDecimal(tb, "TB")
+	case bytes >= GB:
+		gb := float64(bytes) / float64(GB)
+		return formatUnitRoundedUpOneDecimal(gb, "GB")
+	case bytes >= MB:
+		mb := float64(bytes) / float64(MB)
+		return formatUnitRoundedUpOneDecimal(mb, "MB")
+	case bytes >= KB:
+		kb := float64(bytes) / float64(KB)
+		return formatUnitRoundedUpOneDecimal(kb, "KB")
+	default:
+		return fmt.Sprintf("%d bytes", bytes)
```

Comment:
```suggestion
		return fmt.Sprintf("%d B", bytes)
```
nit: to stay consistent with the above